Mac OS failure

Compilation on Mac OS fails (see [here](https://travis-ci.org/blegat/SDPLR.jl/jobs/288588527)) with the following error:
```julia
./proto.h:30:5: error: first parameter of 'main' (argument count) must be of type 'int'
int main(size_t argc, char *argv[]);
```
It seems `size_t` needs to be replaced by `int`.

@sburer is this a known issue ?

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -28,6 +28,7 @@ BinDeps.run(@build_steps begin
         @build_steps begin
             ChangeDirectory(_srcdir)
             `cp Makefile.inc.$osname Makefile.inc`
+            pipeline(`patch -N -p0`, stdin="$rootdir/main_argc.patch")
             @static if is_windows()
                 pipeline(`patch -N -p0`, stdin="$rootdir/windows.patch")
                 `mingw32-make mingw`

--- a/deps/main_argc.patch
+++ b/deps/main_argc.patch
@@ -1,0 +1,24 @@
+--- source/main.c
++++ source/main.c
+@@ -7,7 +7,7 @@
+ #define RANDOM 0
+ // #define DATABLOCKIND(DATA,BLOCK,NUMBLOCK) ((DATA+1)-1)*NUMBLOCK + BLOCK - 1
+ 
+-int main(size_t argc, char *argv[])
++int main(int argc, char *argv[])
+ {
+   size_t h, k, ret, n, nr, *maxranks, *ranks;
+   double *R, *lambda, pieces[8];
+diff --git a/source/proto.h b/source/proto.h
+index fe47b8a..843db2b 100644
+--- source/proto.h
++++ source/proto.h
+@@ -27,7 +27,7 @@ size_t updatelbfgs2(problemdata *data, lbfgsvec *vecs, double *dir, double *grad
+ /* linesearch.c */
+ double linesearch(problemdata *data, double *R, double *D, double max, double *funcval, size_t update);
+ /* main.c */
+-int main(size_t argc, char *argv[]);
++int main(int argc, char *argv[]);
+ size_t getstorage(size_t m, size_t numblk, size_t *blksz, char *blktype, size_t *CAinfo_entptr, size_t *passedn, size_t *passednr, size_t *maxranks);
+ size_t readin(size_t m, size_t numblk, size_t *blksz, char *blktype, double *R, double *lambda, size_t *maxranks, size_t *ranks, double *pieces, FILE *fid);
+ size_t writeout(size_t m, size_t numblk, size_t *blksz, char *blktype, double *R, double *lambda, size_t *maxranks, size_t *ranks, double *pieces, FILE *fid);

